libs/wingc/src/ast.rs

```diff
@@ -316,6 +320,10 @@ pub enum ExprKind {
 		// We're using an ordered map implementation to guarantee deterministic compiler output. See discussion: https://github.com/winglang/wing/discussions/887.
 		fields: BTreeMap<Symbol, Expr>,
 	},
+	JsonLiteral {
+		type_: TypeAnnotation,
```

Comment:
Do we need a `TypeAnnotation` here? I mean a JsonLiteral is always either a `Json` or a `MutJson`, maybe just a `is_mut: bool` is enough here. Will reduce more complex error checks down the line.

Reply:
You are right it was much simpler 😅